### ISSUE TITLE
SuppressWarningsHolder naming inconsistency giving java.lang.ClassNotFoundException

### ISSUE THREAD

**Issue by relentless-pursuit** [checkstyle/checkstyle#13845]
Upon trying to work on enabling example tests for `SuppressWarningsHolderExamplesTests`, I constantly encountered with `java.lang.ClassNotFoundException: com.puppycrawl.tools.checkstyle.checks.annotation.SuppressWarningsHolderCheck`. 

Upon debugging the code, I realised that the code appends 'Check' to the fully qualified name of the class. While, we don't have `SuppressWarningsHolderCheck.java` rather `SuppressWarningsHolder.java`. In the method getFullyQualifiedClassName, a specific block of code highlighted below is doing the append operation. 
```
if (!fullyQualifiedClassName.endsWith("Filter")) {
                fullyQualifiedClassName += "Check";
            }
```

Infact, SuppressWarningsHolder is part of check module and needs to be refactored as `SuppressWarningsHolderCheck.java`.

**Comment by nrmancuso** [comment]
@relentless-pursuit please do:
1. Add a link to the check documentation
2. Add permalink to the referenced code
3. Investigate historical reason for this naming (git blame/ GitHub history can be useful for this)

**Comment by romani** [comment]
> needs to be refactored as SuppressWarningsHolderCheck.java


this is big breaking compatibility for our users to change name of module.
and this module is not a Check, it is bad hack, and one day we will need remove it.

**Comment by relentless-pursuit** [comment]
> > needs to be refactored as SuppressWarningsHolderCheck.java
> 
> this is big breaking compatibility for our users to change name of module. and this module is not a Check, it is bad hack, and one day we will need remove it.

But, the code seems to consider it as a check. If you see my recent PR,  https://github.com/checkstyle/checkstyle/pull/13848, I had explicitly add the fully qualified name in the configuration of xml file. If i just use <module name="SuppressWarningsHolder"/>, it complains of `com.puppycrawl.tools.checkstyle.checks.annotation.SuppressWarningsHolderCheck`. As you mentioned refactoring it to a check class won't be a good idea. But, then the other way around would be to ensure the the the fully qualified name is returned correctly i.e. `com.puppycrawl.tools.checkstyle.checks.annotation.SuppressWarningsHolder`. Meanwhile, I will follow @nrmancuso instructions to add more substatance to the issue.

**Comment by romani** [comment]
We need to extend our test code to handle this unfortunate module in special way, config in example should be same as users have in config.

**Comment by rnveach** [comment]
Code is at https://github.com/checkstyle/checkstyle/blob/c2f5568c26e22b3d40f16346f89cf677371e0ac8/src/test/java/com/puppycrawl/tools/checkstyle/bdd/InlineConfigParser.java#L291-L293 .

Truthfully, `setModuleName` and this method should probably be rewritten to use `PackageObjectFactory` to identify the module as this is our authority on identifying a module. Then we don't have to use any "hacks" and would be able to expand this BDD to other 3rd party repositories, like sevntu. I am pretty sure I brought this up during the BDD GSoC project, but it was never done.

**Comment by relentless-pursuit** [comment]
> Code is at
> 
> https://github.com/checkstyle/checkstyle/blob/c2f5568c26e22b3d40f16346f89cf677371e0ac8/src/test/java/com/puppycrawl/tools/checkstyle/bdd/InlineConfigParser.java#L291-L293
> 
> .
> Truthfully, `setModuleName` and this method should probably be rewritten to use `PackageObjectFactory` to identify the module as this is our authority on identifying a module. Then we don't have to use any "hacks" and would be able to expand this BDD to other 3rd party repositories, like sevntu. I am pretty sure I brought this up during the BDD GSoC project, but it was never done.

I can work on it. Please let me know.

**Comment by romani** [comment]
I highly recommend to extend parser with special handling of such Holder as hardcode. 
Unblock examples PRs.
And in separate PR do better design of class resolution as @rnveach  suggested.
